File: Strings/2042. Check if Numbers Are Ascending in a Sentence/2042. Check if Numbers Are Ascending in a Sentence.py
class Solution:
    def areNumbersAscending(self, s: str) -> bool:
        num = 0
        curr_num = ""
        for char in s:
            if char.isdigit():
                curr_num += char
            else:
                if curr_num != "":
                    if int(curr_num) <= num:
                        return False
                    num = int(curr_num)
                    curr_num = ""

        if curr_num != "":
            if int(curr_num) <= num:
                return False
            num = int(curr_num)
        
        return True

# Time: O(n)
# Space: O(1)

File: Strings/2042. Check if Numbers Are Ascending in a Sentence/2042. Check if Numbers Are Ascending in a Sentence.js
/**
 * @param {string} s
 * @return {boolean}
 */
var areNumbersAscending = function (s) {
  let num = 0;
  let curr_num = "";

  for (const char of s) {
    if (char >= "0" && char <= "9") curr_num += char;
    else {
      if (curr_num !== "") {
        if (Number(curr_num) <= num) return false;
        num = Number(curr_num);
        curr_num = "";
      }
    }
  }

  if (curr_num !== "") {
    if (Number(curr_num) <= num) return false;
    num = Number(curr_num);
  }

  return num;
};

// Time: O(n)
// Space: O(1)
